libft/sources/map/mapss/ft_mapss_ncmp.c

```diff
@@ -33,25 +33,25 @@ int	ft_mapss_ncmp(t_mapss *a, t_mapss *b, int n)
 	t_dlist	*b_node;
 	int		res;
 	int		i;
+	int		maxi;
 
-	i = *ft_min_int(&a->size, &n);
-	n = *ft_min_int(&b->size, &i);
+	if (n == 0)
+		return (0);
+	maxi = *ft_min_int(&b->size, &a->size);
 	i = -1;
 	a_node = a->head;
 	b_node = b->head;
```

Comment:
no check if head == NULL.
do you assume to use this function only if map is not empty?

Reply:
I assume the internal state of map is consistent, and the size field represents the actual quantity of entries. In this case we are iterating to the minimal of two sizes and won't get a nullptr.